docs/framework/20categories/30implementation/audit.md

```diff
@@ -1,10 +1,11 @@
 ### Code Auditing
 
-TODO: This section needs to be re-worked.
+Code audits are necessary to ensure that a protocol’s code performs as per its intended logic. Code that has not been audited is far more likely to contain bugs than code that has been audited. Any protocol that does not have code audits should be trusted less by users and developers. However, code audits should not be viewed as a comprehensive security solution.  For example, protocols sometimes conduct audits on specific parts of their architecture or deploy unaudited code for changes/new features, which can be a potential source of risk outside of the initial audit.
 
-Has the code been audited?
+Thus, when assessing the security of a protocol based on code audits, one must consider the following:
+* Has the code been audited? How many audits have been completed?
+* When was the most recent audit? Has the protocol been upgraded since the last audit?
```

Comment:
Another consideration: Have audits been completed by different organisations? Different auditors will find different issues?

Reply:
Another consideration: What was the scope of the audit(s)? That is, did the audit cover some of the contract code, all of the contract code, all of the contract code and all of the off-chain code, or something else?